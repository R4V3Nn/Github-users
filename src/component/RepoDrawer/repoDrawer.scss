.drawer-items {
	list-style: none;
}

.drawer-items .title {
	padding-right: 10px;
	text-transform: capitalize;
	font-weight: bold;
}
