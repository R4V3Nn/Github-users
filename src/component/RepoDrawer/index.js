import React from 'react';
import { Drawer } from '@talend/react-components';

import './repoDrawer.scss';

const RepoDrawer = props => {
	const { item } = props;

	return (
		<Drawer>
			<h2>Repo informarion:</h2>
			<ul className="drawer-items">
				{
					Object.keys(item).map(key => (
						typeof item[key] === 'string' ?
							<li>
								<span className="title">{`${key}:`} </span>
								<span>{`${item[key]}`}</span>
							</li> : null

					))
				}
			</ul>
		</Drawer>
	);
};

export default RepoDrawer;
