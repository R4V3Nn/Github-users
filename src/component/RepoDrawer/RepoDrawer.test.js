import React from 'react';
import { shallow } from 'enzyme';

import RepoDrawer from './';

describe('RepoDrawer component', () => {
	it('should render RepoDrawer component', () => {
		// when
		const wrapper = shallow(<RepoDrawer />);

		// then
		expect(wrapper.getElement()).toMatchSnapshot();
	});
});
