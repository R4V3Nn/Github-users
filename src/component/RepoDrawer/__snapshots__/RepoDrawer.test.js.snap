// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepoDrawer component should render RepoDrawer component 1`] = `
<Drawer
  footerActions={Object {}}
  title="REPO INFORMATION:"
  withTransition={true}
>
  <Connect(DrawerForm) />
</Drawer>
`;
